frontend/modules/shared/data-access/src/lib/api.params.ts

```diff
@@ -14,7 +14,8 @@ type OnlyKeysOfType<T, TProp> = {
 
 export type ApiFilter<T> = ApiFilterOperator<T, 'eq'> &
   ApiFilterOperator<OnlyKeysOfType<T, string>, 'ct'> &
-  Partial<{ search: string }>;
+  Partial<{ search: string }> &
+  Partial<{ unseen: boolean }>;
```

Comment:
To dodaje, że na każdym filtrze teraz można dodać `{ unseen: boolean }` gdzie ma to sens tylko dla tikcetów. Nie lepiej dopasować endpoint, żeby przyjmował filtr `unseen_eq=true/false` żeby pasowało do wszystkich pozostałych endpointów? Wtedy wystarczy dodać pole `unseen` do `Ticket`

Reply:
czaję czaję lookne czy uda mi się to jakoś łatwo i przyjemnie zrobić

Reply:
Prawdopodobnie miejsce do zmiany to TicketSpecification

Reply:
`TicketService.java:120`

```java
        if (filterVars.containsKey("unseen")) {
            filterVars.remove("unseen");
            combinedSpecification = combinedSpecification.and(new UnseenTicketSpecification(id));
        }
```

Przerobić na `"unseen_eq"` i potem na froncie i jest elegancko wtedy

Reply:
Rozwiązałem to przez podmiankę `unseen` na `unseed_eq` ale no nie podoba mi się to rozwiązanie za bardzo. Wypchniemy że z tym requestem żeby feature notyfikacji był ale chyba jeszcze do tego usiądę i to poprawię.

Unseen nie jest rzeczywistym memberem obiektu Ticket więc gdyby ktoś na froncie chciał zrobić `.unseen` to aplikacja pewnie crashuje XD

Reply:
Zamiast do `Ticket` też możesz spróbować po prostu zrobić `override` na `getAll` serwisu i tam zmienić typ, że `ApiFilter<Ticket> & { unseen?: true }` to wtedy tylko w serwisie to będzie